nifi-nar-bundles/nifi-ocr-bundle/nifi-ocr-processors/src/main/java/org/apache/nifi/processors/ocr/TesseractOCRProcessor.java

```diff
@@ -0,0 +1,355 @@
+            File[] files = getTesseractLanguages(newValue);
+
+            //Guard against creating an empty list of allowable values in case the user points to an invalid directory
+            if (files != null && files.length > 0) {
+                for (int i = 0; i < files.length; i++) {
+                    if (getLogger().isDebugEnabled()) {
+                        getLogger().debug("Found Tesseract supported language: " + files[i].getName());
+                    }
+                    SUPPORTED_LANGUAGES.add(StringUtils.split(files[i].getName(), ".")[0]);
+                }
+            } else {
+                getLogger().debug("No languages found in user specified Tessdata directory: '" + newValue + "'");
+            }
+
+        }
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        //Setup the Tesseract instance once the processor is scheuled
+        tesseract = new Tesseract();
+        tesseract.setLanguage(context.getProperty(TESSERACT_LANGUAGE).getValue());
+        tesseract.setPageSegMode((context.getProperty(TESSERACT_PAGE_SEG_MODE).asInteger()));
+    }
+
+
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        //Transfer the original
+        session.transfer(session.clone(flowFile), REL_ORIGINAL);
+        AtomicBoolean errors = new AtomicBoolean(false);
+
+        FlowFile ff = session.write(flowFile, new StreamCallback() {
+            @Override
+            public void process(InputStream inputStream, OutputStream outputStream) throws IOException {
+                tesseract.setDatapath(context.getProperty(TESS_DATA_PATH).evaluateAttributeExpressions(flowFile).getValue());
+
+                //Builds the list of Tesseract configs.
+                Map<String, String> configs = buildTesseractConfigs(context.getProperty(TESSERACT_CONFIGS).getValue());
+                for (Map.Entry<String, String> entry : configs.entrySet()) {
+                    getLogger().debug("Tesseract Config Key : '" + entry.getKey()
+                            + "' Tesseract Config Value : '" + entry.getValue() + "'");
+                    tesseract.setTessVariable(entry.getKey(), entry.getValue());
+                }
+
+                try {
+                    BufferedImage imBuff = ImageIO.read(inputStream);
+                    outputStream.write(tesseract.doOCR(imBuff).getBytes());
+                } catch (TesseractException te) {
+                    getLogger().error(te.getMessage());
+                    if (te.getCause().getMessage().equals("image == null!")) {
+                        session.transfer(flowFile, REL_UNSUPPORTED_IMAGE_FORMAT);
+                    } else {
+                        session.transfer(flowFile, REL_FAILURE);
+                    }
```

Comment:
@jdye64 I think it would be appropriate to penalize the FlowFile if TesseractException. So basically this:
```
session.transfer(session.penalize(flowFile), REL_FAILURE);
```